datafusion/common/src/types/native.rs

```diff
@@ -0,0 +1,328 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::{ops::Deref, sync::Arc};
+
+use arrow_schema::{DataType, Field, Fields, IntervalUnit, TimeUnit, UnionFields};
+
+use super::{LogicalType, TypeSignature};
+
+/// A record of a native type, its name and its nullability.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct NativeField {
+    name: String,
+    native_type: NativeType,
+    nullable: bool,
+}
+
+impl NativeField {
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    pub fn native_type(&self) -> &NativeType {
+        &self.native_type
+    }
+
+    pub fn nullable(&self) -> bool {
+        self.nullable
+    }
+}
```

Comment:
Why not simply make these fields `pub`?

Reply:
I wanted to use the same access pattern as arrow's Field object but definitely a fair point.

Reply:
Done!